cmd/copyEnumeratorInit_test.go

```diff
@@ -0,0 +1,174 @@
+	p := azblob.NewPipeline(azblob.NewAnonymousCredential(), azblob.PipelineOptions{})
+
+	// List
+	rawBlobURLWithSAS := scenarioHelper{}.getRawBlobURLWithSAS(c, containerName, dirName)
+	blobTraverser := newBlobTraverser(&rawBlobURLWithSAS, p, ctx, true, true, func(common.EntityType) {}, false, common.CpkOptions{}, false, false, false)
+
+	// dir but recursive flag not set - fail
+	cca := CookedCopyCmdArgs{StripTopDir: false, Recursive: false}
+	err := cca.validateSourceDir(blobTraverser)
+	c.Assert(err.Error(), chk.Equals, "cannot use directory as source without --recursive or a trailing wildcard (/*)")
+
+	// dir but recursive flag set - pass
+	cca.Recursive = true
+	err = cca.validateSourceDir(blobTraverser)
+	c.Assert(err, chk.IsNil)
+	c.Assert(cca.IsSourceDir, chk.Equals, true)
+}
+
+func (ce *copyEnumeratorSuite) TestValidateSourceFileExists(c *chk.C) {
+	bsu := getBSU()
+
+	// Generate source container and blobs
+	containerURL, containerName := createNewContainer(c, bsu)
+	defer deleteContainer(c, containerURL)
+	c.Assert(containerURL, chk.NotNil)
+
+	defer deleteContainer(c, containerURL)
+
+	fileName := "source_file"
+	_, fileName = createNewBlockBlob(c, containerURL, fileName)
+
+	ctx := context.WithValue(context.TODO(), ste.ServiceAPIVersionOverride, ste.DefaultServiceApiVersion)
+	p := azblob.NewPipeline(azblob.NewAnonymousCredential(), azblob.PipelineOptions{})
+
+	// List
+	rawBlobURLWithSAS := scenarioHelper{}.getRawBlobURLWithSAS(c, containerName, fileName)
+	blobTraverser := newBlobTraverser(&rawBlobURLWithSAS, p, ctx, true, true, func(common.EntityType) {}, false, common.CpkOptions{}, false, false, false)
+
+	cca := CookedCopyCmdArgs{StripTopDir: false, Recursive: false}
+	err := cca.validateSourceDir(blobTraverser)
+	c.Assert(err, chk.IsNil)
+	c.Assert(cca.IsSourceDir, chk.Equals, false)
+}
+
+func (ce *copyEnumeratorSuite) TestValidateSourceFileDoesNotExist(c *chk.C) {
+	bsu := getBSU()
+
+	// Generate source container and blobs
+	containerURL, containerName := createNewContainer(c, bsu)
+	defer deleteContainer(c, containerURL)
+	c.Assert(containerURL, chk.NotNil)
+
+	defer deleteContainer(c, containerURL)
+
+	fileName := "source_file"
+
+	ctx := context.WithValue(context.TODO(), ste.ServiceAPIVersionOverride, ste.DefaultServiceApiVersion)
+	p := azblob.NewPipeline(azblob.NewAnonymousCredential(), azblob.PipelineOptions{})
+
+	// List
+	rawBlobURLWithSAS := scenarioHelper{}.getRawBlobURLWithSAS(c, containerName, fileName)
+	blobTraverser := newBlobTraverser(&rawBlobURLWithSAS, p, ctx, true, true, func(common.EntityType) {}, false, common.CpkOptions{}, false, false, false)
+
+	cca := CookedCopyCmdArgs{StripTopDir: false, Recursive: false}
+	err := cca.validateSourceDir(blobTraverser)
+	c.Assert(err.Error(), chk.Equals, common.FILE_NOT_FOUND)
+	c.Assert(cca.IsSourceDir, chk.Equals, false)
+}
+
+func (ce *copyEnumeratorSuite) TestValidateSourceWithWildCard(c *chk.C) {
+	bsu := getBSU()
```

Comment:
Din't get which part is putting WildCard Filter here.

Reply:
StripTopDir is what's set on a wildcard